### ISSUE TITLE
[BUG / possible BUG] Can't extract CRLs when using OpenSSL 3.X

### ISSUE THREAD

**Issue by dcooper16** [drwetter/testssl.sh#2147]
**Command line / docker command to reproduce**
testssl.sh --server-defaults --openssl=\<path to OpenSSL 3.0.2\>/openssl google.com

**Expected behavior**
The Certificate Revocation List lines for the two certificates should not be empty.

**Your system (please complete the following information):**
 - OS: Ubuntu 20.04.4 LTS
 - Platform: Linux 5.13.0-39-generic x86_64
 - Version: 3.1dev
 - Version if running from git repo: e82178719d7f17734ea1585be5e0476ea0117b89

**Additional context**
The way that OpenSSL pretty prints certificates has changed and some blank lines that were previously in the pretty printed output are no longer there. (Below is the output from OpenSSL 1.1.1 and OpenSSL 3.0.2, showing just the relevant part of the output.)

The code to extract CRLs from the certificate is as follows:
```
crl="$(awk '/X509v3 CRL Distribution/{i=50} i&&i--' <<< "$cert_txt" | awk '/^$/,/^            [a-zA-Z0-9]+|^    Signature Algorithm:/' | awk -F'URI:' '/URI/ { print $2 }')"
```
With OpenSSL 3.0.2, the second `awk` in this line outputs an empty string, since it appears to be looking for a blank line to come after "X509v3 CRL Distribution Points". There may be a way to replace the calls to `awk` with the use of Bash string manipulation, but I don't understand `awk` well enough to fix the current version.

OpenSSL 1.1.1:
```
            X509v3 CRL Distribution Points: 

                Full Name:
                  URI:http://crls.pki.goog/gts1c3/QqFxbi9M48c.crl

            CT Precertificate SCTs:
```
OpenSSL 3.0.2:
```
            X509v3 CRL Distribution Points: 
                Full Name:
                  URI:http://crls.pki.goog/gts1c3/QqFxbi9M48c.crl
            CT Precertificate SCTs:
```

**Comment by drwetter** [comment]
Hi David.

good catch!

The awk pattern is not good in the first place. It should rather search for ~ _Full Name\nURL:_ immediately after _X509v3 CRL Distribution Points_.  And, yes, awk is not good style anymore here, it could be e.g. either _read_ or multiline bash which we have some where.

**Comment by dcooper16** [comment]
Hi Dirk,

I tried coming up with a way to do this using just Bash string manipulation, but so far have not been successful. What seems to work is the following:
```
crl="$(awk '/X509v3 CRL Distribution/{i=50} i&&i--' <<< "$cert_txt" | awk '/^$|^.*Name.*$|^.*Reasons.*$|^.*CRL Issuer.*$/,/^            [a-zA-Z0-9]+|^    Signature Algorithm:/' | awk -F'URI:' '/URI/ { print $2 }')"
```
Unless a CRL Distribution Points extension is malformed (e.g., it is empty), it will have to include at least of the following lines: "Full Name", "Relative Name", "Reasons", or "CRL Issuer". So, I just changed the `awk` to start with any of these lines. I tested it with several versions of OpenSSL and LibreSSL and it worked with all of them except OpenSSL 0.9.8zh. The current version doesn't work with OpenSSL 0.9.8zh either, so this is not a regression.

Searching for "Full Name\nURI" should work in most cases. However, there is no guarantee that this string will appear in the CRL Distribution Points extension. While the string unlikely to appear in any other extension, there is a small risk that it would.
